ts/components/services/ServiceListItem.tsx

```diff
@@ -42,16 +53,35 @@ const styles = StyleSheet.create({
   }
 });
 
-export class ServiceListItem extends React.PureComponent<Props> {
+export class ServiceListItem extends React.PureComponent<Props, State> {
   public render() {
+    const { isLongPressModeEnabled } = this.props;
     const potService = this.props.item;
     const enabledChannels = pot.map(potService, service =>
       getEnabledChannelsForService(this.props.profile, service.service_id)
     );
 
-    const onPress = pot.toUndefined(
-      pot.map(potService, service => () => this.props.onSelect(service))
-    );
+    if (pot.isSome(potService)) {
+      // tslint:disable-next-line no-object-mutation
+      this.state = {
+        uiEnabledChannels: getEnabledChannelsForService(
+          this.props.profile,
+          potService.value.service_id
+        )
+      };
+    }
+
+    const onPress = !this.props.isLongPressModeEnabled
+      ? pot.toUndefined(
+          pot.map(potService, service => () => this.props.onSelect(service))
+        )
+      : undefined;
+
+    const onSwitchTap = this.props.isLongPressModeEnabled
```

Comment:
Shouldn't `this.props.isLongPressModeEnabled` always be true?

Reply:
Yes :)